### ISSUE TITLE
Truncated lifting on ARM

### ISSUE THREAD

**Issue by kereoz** [angr/pyvex#74]
(Relevant binary attached)
The following block is truncated:
```
addr = p.loader.main_bin.get_symbol('eth_send').addr+0x7c+p.loader.main_bin.rebase_addr
In [11]: p.factory.block(addr).pp()
0x4007ec:	pop	{r3, pc}
```

This block should be:

```
7c:   e8bd8008        pop     {r3, pc}
80:   e3e00000        mvn     r0, #0
84:   e8bd8008        pop     {r3, pc}
```

It is worth nothing that the rest of the block is decoded correctly when skipping the first instruction:

```
In [7]: p.factory.block(addr+4).pp()
0x4007f0:	mvn	r0, #0
0x4007f4:	pop	{r3, pc}
```

At first glance, everything looks fine in terms of size calculation until the call to pyvex.IRSB in simuvex/engines/vex/engine.py so I assume this bug comes from pyvex or below.

[eth.o.gz](https://github.com/angr/simuvex/files/1009358/eth.o.gz)

**Comment by zardus** [comment]
My guess is that VEX is interpreting that first instruction as a return (which makes sense -- it pops into the program counter). Technically, that's where the basic block ends.

**Comment by kereoz** [comment]
Back to this (sorry about the delay!). Please ignore the above nonsense :) (got things mixed up / wrong blocks). I initially meant that the block at 0x80 does not embed the literal pool that follows it (which is the case  whenever a block is followed by a literal pool), and in that sense, I considered it was truncated since I was expecting this literal pool to be lifted as well as part of the same block. That said, I guess this only makes sense if we make the (arbitrary) assumption that literal pools should be represented as part of the last block of each section (which is more or less where the compiler inserts them, although they are technically inserted after the last block). To clarify, this is not an issue per se, but more the question of how to represent literal pools at the basic block  level, and if it makes sense to do it at all (as opposed to leave it for "higher-level" analyses to figure it out). They're not code, but they are part of the text segment, and in some use cases (e.g., lightweight analyses modifying the text segment), it is useful to have them represented without having to analyze memory access patterns. Now, for analyses that actually "execute" the blocks, it doesn't matter, and the current behavior is fine.

Bottom line: if that sounds like something that could be useful to anyone else, is it better to 1) implement a (heuristic-based) way to represent literal pools at the pyvex block level, or 2) decide to leave it for higher-level analyses? (And perhaps add a CFG node property for this? But we might not always recover everything this way).

**Comment by kereoz** [comment]
Closing that issue, we had a discussion with Yan and determined that the current behavior is consistent other similar decisions made in elsewhere in the lifting. 
People lifting ARM blocks as part of static analyses should watch out for literal pools and manually add missing constants.
